core/templates/pages/story-editor-page/services/story-editor-state.service.ts

```diff
@@ -128,11 +138,15 @@ angular.module('oppia').factory('StoryEditorStateService', [
        * lifetime of the editor. This function never returns null, though it may
        * return an empty story object if the story has not yet been
        * loaded for this editor instance.
+       * @return {Story} Returns the current story to be shared.
        */
       getStory: function() {
         return _story;
       },
 
+      /**
+       * @return {object} Returns an array of the list of skill summaries.
```

Comment:
What would you suggest is the best way to find an interface in this situation? Would it be best to run the application and search through there or do you have any tips? In this case, would the object be a SkillSummary? When you say to be more exact, instead of object should i say array of x?

Reply:
Yep, in general, running the application and verifying the type (using console logs or debugger tool) would be a good way to do it. Also, you might find "Deducing variable type from the code" section in this [comment](https://github.com/oppia/oppia/issues/8668#issue-566518484) useful for your question. I think @nishantwrp would be able to give you more tips on this since he has been working on typing related stuff recently.

Reply:
Hi @pengcheng92, yup you should try to deduce the type from code. The best way is to check what values are actually passed in this function. Looking at that you'll `EditableStoryBackendApiService` is used to fetch the story. And then looking at the backend controller you can easily find the correct type of skill summary. In this case which is `SkillSummaryBackendDict` in the `skill-summary-object.factory.ts`. (This file is not present in your branch, you'll need to merge from develop once.)

Reply:
Thanks for the tips. Updated!